indico/modules/events/contributions/operations.py

```diff
@@ -153,3 +155,29 @@ def delete_subcontribution(subcontrib):
     logger.info('Subcontribution %s deleted by %s', subcontrib, session.user)
     subcontrib.event_new.log(EventLogRealm.management, EventLogKind.negative, 'Subcontributions',
                              'Subcontribution "{}" has been deleted'.format(subcontrib.title), session.user)
+
+
+@no_autoflush
+def create_contribution_from_abstract(abstract, contrib_session=None):
+    event = abstract.event_new
+    contrib_person_links = set()
+    person_link_attrs = {'_title', 'address', 'affiliation', 'first_name', 'last_name', 'phone', 'author_type',
+                         'person', 'is_speaker'}
+    for abstract_person_link in abstract.person_links:
+        link = ContributionPersonLink()
+        link.populate_from_attrs(abstract_person_link, person_link_attrs)
```

Comment:
Set the person first (pass it in the constructor). Setting the other attributes requires a proper person association (because of the `override_attr` logic used in person links).